starknet_devnet/blueprints/rpc/structures/responses.py

```diff
@@ -200,8 +195,10 @@ async def txn_type() -> RpcTxnType:
         "transaction_hash": rpc_felt(txr.transaction_hash),
         "actual_fee": rpc_felt(txr.actual_fee or 0),
         "status": status(),
-        "block_hash": rpc_felt(txr.block_hash) if txr.block_hash is not None else None,
-        "block_number": txr.block_number or None,
+        "block_hash": rpc_felt(txr.block_hash)
+        if txr.block_hash is not None
+        else rpc_felt(0),
+        "block_number": txr.block_number if txr.block_number is not None else 0,
```

Comment:
Spec doesn't allow nulls for block_hash and number even for rejected transactions. 0 seemed like reasonable default but I'm unsure.

Reply:
When would `txr.block_hash` even be `None`?

Reply:
Not sure honestly, but in the cairo-lang `TransactionInBlockInfo` both block hash and number are typed as `Optional[int]`.

Reply:
Actually this could be the case if the tx is a part of the currently pending block (it doesn't yet have a block hash).